stl/inc/chrono

```diff
@@ -5486,34 +5500,60 @@ struct formatter<_CHRONO day, _CharT> {
         }
 
         for (const auto& _Spec : _List) {
-            if (_Spec._Type != '\0' && _Spec._Type != 'd' && _Spec._Type != 'e') {
-                _THROW(format_error("Invalid type for type day."));
-            }
-            if (_Spec._Modifier != '\0' && _Spec._Modifier != 'O') {
-                _THROW(format_error("Invalid modifier for type day."));
+            if (_Spec._Type != '\0' && _RANGES find(_Valid_types, _Spec._Type) == _Valid_types.end()) {
+                _THROW(format_error("Invalid type."));
             }
+            _Check_modifier(_Spec._Type, _Spec._Modifier);
         }
 
         return _Res_iter;
     }
 
-    template <class _FormatContext>
-    typename _FormatContext::iterator format(const _CHRONO day& _Val, _FormatContext& _FormatCtx) {
-        using _Ctx_iter = typename _FormatContext::iterator;
+    void _Check_modifier(const _CharT _Type, const _CharT _Modifier) {
+        if (_Modifier == _CharT{'\0'}) {
+            return;
+        }
+
+        enum _Allowed_bit { _E_mod = 1, _O_mod = 2, _EO_mod = _E_mod | _O_mod };
+
+        struct _Table_entry {
+            char _Type;
+            _Allowed_bit _Allowed;
+        };
+
+        const _Allowed_bit _Mod = _Modifier == _CharT{'E'} ? _E_mod : _O_mod;
+
+        static const _Table_entry _Table[] = {{'d', _O_mod}, {'e', _O_mod}, {'m', _O_mod}};

```

Comment:
`const` versus `constexpr` doesn't affect performance (`constexpr` insists that the initializers be constant expressions so it's better in that respect). However, `static` significantly increases performance. The issue is that if you have a non-static array (whether `const` or `constexpr`), its elements are supposed to have distinct addresses from any other objects in the program. The compiler can optimize things away if it can prove that nothing else can observe it - however, arrays are inherently address-taken for virtually all use. Therefore, non-static arrays lead to the compiler emitting codegen to push every individual value onto the stack! :scream_cat: In contrast, a static array can be burned into read-only memory, with a unique address for the program's lifetime.

Scalar variables don't benefit from `static` (unless they're address-taken somehow), so we conventionally don't mark them as `static constexpr`.